### ISSUE TITLE
DatetimeEncored can add holiday/weekend binary features

### ISSUE THREAD

**Issue by baggiponte** [skrub-data/skrub#710]
### Problem Description

Add weekends/holiday features to the data

### Feature Description

The DatetimeEncoder could enjoy a new `add_weekends` param that generates a binary dummy variable that is set to 1 when the day is a weekend day.

A `holidays: str | None = None` param could accept a world country supported by the [holidays package](https://github.com/vacanza/python-holidays) and generate another binary feature col. It could technically be its own encoder (HolidayEncored) if you think the DateTimeEncoder is doing too much.

**Comment by Vincent-Maladiere** [comment]
Hey @baggiponte, thanks for the suggestion!

1. The weekend binary variable would bring value and is straightforward to implement!
2. The holidays package would add a new runtime dependency, and we prefer avoiding that.

Instead, we could improve one of our existing examples (and preferably increase its classification or regression score) using the holidays package as a documentation dependency. This way, we would demonstrate its usefulness while showcasing its usage with skrub.

Would you be interested in implementing one of those (or both)?

**Comment by Vincent-Maladiere** [comment]
1. Yes indeed, the documentation is being fixed, but you can access the source code file you mentioned.

	This file is actually generated via [nbconvert](https://nbconvert.readthedocs.io/en/latest/), we sync a notebook to a python file, so that changes in the notebook are reflected in the file. We then manually process it to make it "sphinx compatible", i.e. we replace `#%` with `#######` for cell breaks.

	**Therefore, I suggest you reproduce example 03 in your own notebook, make the appropriate changes, and put these changes into the source file, trying to respect the sphinx syntax**. I'll help you debug it if needed.

2. Seems like a good idea to use a light `FunctionTransformer`, go ahead!

**Comment by rcap107** [comment]
Hello! We talked about holidays as features during this week's meeting, and I found this API that might be useful for adding holidays as features: https://api-ninjas.com/api/holidays

Of course this would mean relying on an external API (maybe problematic), and that the user should choose the proper calendar for the dataset they're working with (e.g., if the dataset is about French data, no point in using German holidays).

Another issue is that some holidays like Easter do not fall on fixed dates, which is what an API like the one I linked would be useful for. 

In general, I think that even a set of fixed calendars to use for binary features may be useful to have. Then, it would fall on the user to choose which calendar they think is most relevant to the data.

**Comment by jeromedockes** [comment]
thanks! there is also [python-holidays](https://python-holidays.readthedocs.io/en/latest/)

**Comment by jeromedockes** [comment]
Let's try to revive this, with python-holidays :)
also it might be interesting to add weather eg with https://openweathermap.org/api as suggested by @ogrisel

**Comment by baggiponte** [comment]
Hello there, I've totally neglected this. I'm quite busy right now, but for $CLIENT I did exactly the `FunctionTransformer` + `holidays` trick, so it's just a matter of writing it down. Won't have much time until the end of the month, tough.

**Comment by MarieSacksick** [comment]
A note about the week-ends: it's not actually the same everywhere in the world. As far as i know, in some muslim countries such as in Saudi Arabia, it's Friday and Saturday.

**Comment by rcap107** [comment]
> A note about the week-ends: it's not actually the same everywhere in the world. As far as i know, in some muslim countries such as in Saudi Arabia, it's Friday and Saturday.

Yes, that's very true. I think at the moment we're using Saturday and Sunday as default. Maybe we could change that 🤔 at the moment the parameter is a boolean, we could change it to a string ("friday" or "saturday" for the first day of the weekend), or None to disable it.

**Comment by jeromedockes** [comment]
the weekend is not really the interesting part because it does not add much information compared to day of the week. what is really additional information is holidays, which of course depends a lot on the country and context. python-holidays contains bank holidays for many countries so that would be a start

**Comment by gabrielapgomezji** [comment]
Hello, I'm going to start working on this
